src/datalogger.cpp

```diff
@@ -715,9 +715,21 @@ void Datalogger::setConfiguration(cJSON *config)
   storeDataloggerConfiguration();
 }
 
+// TODO: can I modify this to return a JSON instead? It can then be used when writing the comments before the header in the csv
 void Datalogger::getConfiguration(datalogger_settings_type *dataloggerSettings)
```

Comment:
Good thought, looks like currently this is only used to return the data structure, and then allow the CLI code to translate that into JSON.  I think this was a good initial idea - to separate JSON parsing/creation from the datalogger and sensor drivers - but in fact we already do the JSON for the sensors within the sensor drivers.   So I think it's fine to go ahead and rename this function as getConfigurationJSON and return JSON instead.  The knowledge of JSON structure for a particular class can be contained completely by that class, I think that works nicely.